lib/rails-properties/base.rb

```diff
@@ -12,11 +12,7 @@ def properties(var)
           raise ArgumentError unless var.is_a?(Symbol)
           raise ArgumentError.new("Unknown key: #{var}") unless self.class.default_properties[var]
 
-          if RailsProperties.can_protect_attributes?
-            property_objects.detect { |s| s.var == var.to_s } || property_objects.build({ :var => var.to_s }, :without_protection => true)
-          else
-            property_objects.detect { |s| s.var == var.to_s } || property_objects.build(:var => var.to_s, :target => self)
-          end
+          cached_properties[var] ||= property_objects.find_or_initialize_by(var: var)
```

Comment:
This memoization is being done to support new property objects. If a property object isn't in the database, `property_objects.find_or_initialize_by(var: var)` returns an new object in memory. Since we can write multiple values to a new property like

```ruby
account.properties(:portal).premium = true
account.properties(:portal).fee = 42.5
```

`#properties` needs to return the same instance of the property object. Otherwise, we'll have two different property object for the same `var` (`:portal` in the above example), and [that's not allowed](https://github.com/1debit/rails-properties/blob/e97fe12249e065c9d8f93c0185f441262c55d94b/lib/generators/rails_properties/migration/templates/migration.rb#L15).

This memoization was happening implicitly before when everything was being loading into memory.